### PR TITLE
Basic fixes for the Syndicate Reinforcement Teleporter

### PR DESCRIPTION
**Pull request by Mervill**
## About the PR 

I ran outta motivation on this one. I did some basic fixes but there's still some longer-term problems with the Syndicate Reinforcement Teleporter that need to be addressed.

reinforcement

**Changelog**

:cl:
- fix: The Syndicate Reinforcement Teleporter creates a real syndicate member complete with objectives and codewords.
- fix: The Syndicate Reinforcement Teleporter creates a syndicate member of any race instead of just humans.

### CODE REVIEW

**Comment by ZeroDayDaemon** [comment]
Do they get the objectives of who uses it?

**Comment by Mervill** [comment]
no they get random new objectives like a fresh traitor. making the objectives smarter is definitely a good idea but not in the cards for me at the moment

**Comment by Elijahrane** [comment]
so you could theoretically buy a reinforcement and the reinforcement gets an objective to kill you?

**Comment by EmoGarbage404** [comment]
> so you could theoretically buy a reinforcement and the reinforcement gets an objective to kill you?

Incredibly based and funny.

**Comment by Mervill** [comment]
are traitors not excluded from the list of possible kill traitors?

**Comment by mirrorcult** [comment]
> are traitors not excluded from the list of possible kill targets?

no, and they shouldnt be excluded, really

**Comment by GoodWheatley** [comment]
I said this in ideamen and I’ll say it here again, I really don’t like the idea of the Reinforcement Teleporter creating an entirely new traitor. Most of the value of the item was that you had a partner to help complete your objectives, and this change invalidates this value. And keep in mind that this is also nukie-purchaseable, which could lead to some weird conflicts where the reinforcement is told to kill a nukie.

Other changes look good though!

**Comment by HoidC** [comment]
Wasn't the point of this for nuclear operatives, not traitors? And if it is for traitors, why does it create another traitor with different objectives?

**Comment by ZeroDayDaemon** [comment]
Ideally this would copy the objectives of the user if they are an antag (that way works for both nukies and non nukies) and if a non antag uses it, it should spawn an antag with fresh objectives (to prevent any sane person from using it, also if it refused on non antags it would become a validhunters dream)

**Comment by GoodWheatley** [comment]
After thinking about this for a while, the traitor teleporter and nukie teleporter should be separate items. Right now buying a nukie reinforcement means you’ll have to buy a hardsuit for them, which only adds to the cost. They should come equipped with standard nukie gear, and the only way I can see that work is if the teleporters are separate items each spawning in their own unique role.

**Comment by Elijahrane** [comment]
just give them the 'keep other traitor alive' objective with the user as the target

**Comment by GoodWheatley** [comment]
> just give them the 'keep other traitor alive' objective and the 'ensure other traitor achieves at least half their objectives' objective with the user as the target

what if the traitor has the die objective?

**Comment by mirrorcult** [comment]
then it's very funny and they shouldn't have bought a reinforcement

**Comment by Mervill** [comment]
The Objectives thing is a real issue but...
- Funny
- To do that fix properly, I think, needs a new toolbox feature (specifically system-bound data, ask me about it later)
- I am not going to be pushing any more changes to this in a short term, the holidays are coming up and Dwarf Fortress comes out on Tuesday

File: Content.Server/Ghost/Roles/Components/GhostRoleTraitorReinforcementComponent.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+using Content.Server.Ghost.Roles.Events;
+using Content.Server.GameTicking.Rules;
+using Content.Server.Humanoid.Systems;
+using Content.Shared.Humanoid.Prototypes;
+
+namespace Content.Server.Ghost.Roles.Components
+{
+    [RegisterComponent, ComponentReference(typeof(GhostRoleComponent))]
+    public sealed class GhostRoleTraitorReinforcementComponent : GhostRoleComponent
```
Comment (AJCM-git) [inline Content.Server/Ghost/Roles/Components/GhostRoleTraitorReinforcementComponent.cs:14]: Please dont expand E/C code like this, i know GhostTakeoverAvailableComponent does this but that thing is incredibly old. If you want to do this, ECSing GhostRoleComponent is a requirement

**Review by AJCM-git — CHANGES_REQUESTED** [review]

**Comment by AJCM-git** [comment]
closing this as it would require a conversion of E/C code elsewhere and its been inactive. Do reopen once you have time

**Pull request closed without merging**